lib/katello/engine.rb

```diff
@@ -99,6 +89,17 @@ class Engine < ::Rails::Engine
                         #{Katello::Engine.root}/app/lib/headpin/actions
                         #{Katello::Engine.root}/app/lib/katello/actions)
       ForemanTasks.dynflow.config.eager_load_paths.concat(action_paths)
+      ForemanTasks.dynflow.eager_load_actions!
+    end
+
+    # make sure the Katello plugin is initialized before `after_initialize`
+    # hook so that the resumed Dynflow tasks can rely on everything ready.
+    initializer 'katello.register_plugin', :before => :finisher_hook, :after => 'foreman_remote_execution.register_plugin' do
+      ::Foreman::AccessControl::Permission.prepend ::Katello::Concerns::PermissionExtensions
+      require 'katello/plugin'
+
+      # extend builtin permissions from core with new actions
+      require 'katello/permissions'
```

Comment:
I am confused by the moving around of this code. It appears to not change anything other than where it appears in the file?

Reply:
Yes. From that perspective there are no changes at all. But the order does affects if the needed constants are available in plugin registration block.